DirigoEdge/Business/ADAService.cs

```
@@ -0,0 +1,164 @@
+            {
+                return JsonConvert.DeserializeObject<ADAResult>(page.ADAJson);
+            }
+
+            var url = Helper.GetCanonical(page);
+
+            var resp = RunComplainceCheck(url);
+            SaveADAPageResult(pageId, resp.RawResponse.Content, resp.Result.Categories.Error.Count);
+
+            return resp.Result;
+        }
+
+        public ADAResult CheckBlogAdaCompliance(Int32 id)
+        {
+            var blog = Context.Blogs.First(d => d.BlogId == id);
+
+            if (!IsNullOrWhiteSpace(blog.ADAJson) && blog.ErrorCount.HasValue)
+            {
+                return JsonConvert.DeserializeObject<ADAResult>(blog.ADAJson);
+            }
+
+            var url = blog.Canonical;
+
+            var resp = RunComplainceCheck(url);
+            SaveADABlogResult(id, resp.RawResponse.Content, resp.Result.Categories.Error.Count);
+
+            return resp.Result;
+        }
+
+        public void SaveADAPageResult(int pageId, string json, int? errorCount)
+        {
+            if (IsNullOrEmpty(json) || errorCount == null)
+            {
+                return;
+            }
+
+        var page = Context.ContentPages.First(d => d.ContentPageId == pageId);
+
+            page.ErrorCount = errorCount;
+            page.ADAJson = json;
+
+            Context.SaveChanges();
+        }
+
+        public void SaveADABlogResult(int id, string json, int? errorCount)
+        {
+            if (IsNullOrEmpty(json) || errorCount == null)
+            {
+                return;
+            }
+
+            var blog = Context.Blogs.First(d => d.BlogId == id);
+
+            blog.ErrorCount = errorCount;
+            blog.ADAJson = json;
+
+            Context.SaveChanges();
+        }
+
+        public ADAResponse RunComplainceCheck(String url)
+        {
+            var resp = new ADAResponse();
+
+            if (url.Contains(".local/"))
+            {
+                url = "http://www.dirigodev.com";

```

This should either throw an error or at least be configurable.